core/trino-main/src/main/java/io/trino/execution/buffer/LazyOutputBuffer.java

```diff
@@ -209,7 +208,7 @@ public ListenableFuture<BufferResult> get(OutputBufferId bufferId, long token, D
         if (outputBuffer == null) {
             synchronized (this) {
                 if (delegate == null) {
-                    if (stateMachine.getState() == FINISHED) {
+                    if (stateMachine.getState().isTerminal()) {
```

Comment:
We should not return `complete` when the buffer is not `FINISHED`

Reply:
You are correct, fixing in next revision.